IS0801Test.py

```diff
@@ -12,12 +12,405 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from GenericTest import GenericTest
+from GenericTest import GenericTest, NMOSTestException
+from TestHelper import compare_json
+from TestResult import Test
+from NMOSUtils import NMOSUtils
+from is08 import Action, Activation
+from is08.inputs import getInputList
+from is08.outputs import getOutputList
+from is08.active import Active
+from is08.io import IO
+from is08.testConfig import globalConfig
+import time
+import copy
+
+MAPPING_API_KEY = "channelmapping"
 
 
 class IS0801Test(GenericTest):
     """
     Runs IS-08-01-Test
     """
+
     def __init__(self, apis):
         GenericTest.__init__(self, apis)
+        globalConfig.apiUrl = apis[MAPPING_API_KEY]['url']
+        globalConfig.testSuite = self
+        globalConfig.apiKey = MAPPING_API_KEY
+
+    def test_01(self):
+        """Content of the /io view matches resources elsewhere in the API"""
+        test = Test("Content of the /io view matches resources elsewhere in the API")
+        globalConfig.test = test
+
+        inputList = getInputList()
+        outputList = getOutputList()
+        ioInstance = IO()
+        ioJSON = ioInstance.getIOAsJSON()
+
+        mockIoResource = {"inputs": {}, "outputs": {}}
+
+        for input in inputList:
+            mockIoResource["inputs"][input.id] = input.assembleInputObject()
+
+        for output in outputList:
+            mockIoResource["outputs"][output.id] = output.assembleOutputObject()
+
+        if compare_json(mockIoResource, ioJSON):
+            return test.PASS()
+        else:
+            return test.FAIL("IO Resource does not correctly reflect the API resources")
+
+    def test_02(self):
+        """Immediate activation can be called on the API"""
+        test = Test("Immediate action can be called on the API")
+        globalConfig.test = test
+
+        outputList = getOutputList()
+        testRouteAction = outputList[0].findAcceptableTestRoute()
+        activation = Activation()
+        activation.addAction(testRouteAction)
+        activation.fireActivation()
+
+        activeResource = Active()
+        activeResource.assertActionCompleted(testRouteAction)
+        return test.PASS()
+
+    def check_delayed_activation(self, activationTime, activationType):

```

Comment:
Minor detail, but could non-'test_' methods be put at one end of the file? I realise that's not consistent everywhere yet, but I think it makes navigation of the source code a little easier.

Reply:
Fixed